python/deltalake/table.py

```diff
@@ -258,6 +259,15 @@ def __init__(
             log_buffer_size=log_buffer_size,
         )
         self._metadata = Metadata(self._table)
+        
+        if lazy_load:
+            self._table = RawDeltaTable.load_lazy(table_uri)
```

Comment:
How about the storage options here? We probably want to lazy load also tables in cloud storage